File: ISL.ReIdentification.Core/Models/Foundations/AccessAudits/AccessAudit.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System;

namespace ISL.ReIdentification.Core.Models.Foundations.AccessAudits
{

    public class AccessAudit : IKey, IAudit
    {
        public Guid Id { get; set; }
        public string PseudoIdentifier { get; set; }
        public Guid EntraUserId { get; set; }
        public string GivenName { get; set; }
        public string Surname { get; set; }
        public string Email { get; set; }
        public string Purpose { get; set; }
        public string Reason { get; set; }
        public string Organisation { get; set; }
        public bool HasAccess { get; set; }
        public string Message { get; set; }
        public string CreatedBy { get; set; }
        public DateTimeOffset CreatedDate { get; set; }
        public string UpdatedBy { get; set; }
        public DateTimeOffset UpdatedDate { get; set; }
    }
}

File: ISL.ReIdentification.WebApps.Portals/ISL.ReIdentification.Portals.Client/src/models/accessAudit/accessAudit.ts
export class AccessAudit {
    public id: string = "";
    public requestId: string = "";
    public pseudoIdentifier: string = "";
    public entraUserId: string = "";
    public givenName: string = "";
    public surname: string = "";
    public email: string = "";
    public reason: string = "";
    public organisation: string = "";
    public hasAccess: boolean = false;
    public message: string = "";
    public auditType: string = "";
    public transactionId: string = "";
    public count: number = 0
    public createdBy?: string;
    public createdDate?: Date | undefined;
    public updatedBy?: string;
    public updatedDate?: Date | undefined;
}